src/main/scala/org/constellation/rollback/RollbackLoader.scala

```diff
@@ -29,6 +30,21 @@ class RollbackLoader(
     Try(deserializeFromFile[GenesisObservation](genesisObservationPath))
       .map(Right(_))
       .getOrElse(Left(CannotLoadGenesisObservationFile(genesisObservationPath)))
+}
+
+object RollbackLoader{
+
+  private def deserializeFromFile[T](path: String): T =
+    KryoSerializer.deserializeCast[T](File(path).byteArray)
+
+  private def deserializeAllFromDirectory[T](directory: String): Seq[T] =
+    getListFilesFromDirectory(directory).map(s => deserializeFromFile[T](s))
+
+  private def getListFilesFromDirectory(directory: String): Seq[File] =
+    File(directory).list.toSeq
+
+  private def deserializeFromFile[T](file: File): T =
+    KryoSerializer.deserializeCast(file.byteArray)
 
   def loadSnapshotInfoSer(snapshotInfoDir: String): SnapshotInfoSer = {
```

Comment:
`loadSnapshotInfoSer` should be either **private** in `class RollbackLoader` if rollback needs it. Or it shouldn't be there if it's used somewhere else. It clearly breaks the single responsibility principle.

Reply:
Sure, I take back my point above. Lets put it on the SnapshotInfoSer companion object